### DISCUSSION TITLE
Pyaudio - no default output device

### DISCUSSION THREAD

**Discussion by GiacomoMaletto** [termux/termux-packages#11526] [Q&A] [answered]
Hello, I installed first portaudio (19.07.00), then pyaudio (0.2.12). I'm trying to generate a simple sine wave sound on the terminal using code from [here](https://stackoverflow.com/questions/8299303/generating-sine-wave-sound-in-python).
But when I run the code, it says
`OSError: [Errno -9996] Invalid ouput device (no default output device)`
How can I set this so that audio is played?

**Comment by tomty89** [comment, marked as answer]
PortAudio does not support any host API available on Android (i.e. OpenSLES / aaudio). It does not support PulseAudio (which on Termux consists of downstream modules on that supports the aforementioned APIs) or oboe the wrapper either.

And our portaudio build isn't even built with support in ALSA and jack, so you can't use either host API with our build even if your device is rooted (which might allows you to open a sound device directly / exclusively).

Therefore I have no idea why @xtkoba added the package in the first place. (No reason given in #10426 either.)

P.S. Do you actually want to generate a sine wave and play it on the fly or do you really just want a file? You might be merely using the wrong stuff for your goal.

**Reply by GiacomoMaletto** [reply]
Thanks, this is what I needed to know (and saved me from wasting a lot of time).
